graphistry/PlotterBase.py

```diff
@@ -2298,20 +2298,37 @@ def spanner_init(self: Plottable, spanner_config: Dict[str, str]) -> Plottable:
         :return: Plottable with a Spanner connection 
         :rtype: Plottable
         :raises ValueError: If any of the required keys in `spanner_config` are missing or have invalid values.
-
+        :raises SpannerConnectionError: If the connection to Spanner fails.
         """
         from .plugins.spannergraph import SpannerGraph
 
         res = copy.copy(self)
 
-        res._spannergraph = SpannerGraph(res, spanner_config)
+        res._spannergraph = SpannerGraph.from_config(spanner_config)
         logger.debug("Created SpannerGraph object: {res._spannergraph}")
         return res
 
     def kusto_init(self: Plottable, kusto_config: Dict[str, str]) -> Plottable:
+        """
+        Initializes a KustoGraph object with the provided configuration and connects to the cluster
+
+        kusto_config dict must contain the include the following keys:
+            - "cluster": The Kusto cluster name.
+            - "database": The Kusto database name.
+            - "client_id": The Kusto client ID.
+            - "client_secret": The Kusto client secret.
+
+        :param kusto_config: A dictionary containing the Kusto configuration. 
+        :type (Dict[str, str])
+
+        :returns: Plottable with a Kusto connection 
+        :rtype: Plottable
+        :raises ValueError: If any of the required keys in `kusto_config` are missing or have invalid values.
+        :raises KustoConnectionError: If the connection to Kusto fails.
+        """
         from .plugins.kustograph import KustoGraph
         res = copy.copy(self)
-        res._kustograph = KustoGraph(res, kusto_config)
+        res._kustograph = KustoGraph.from_config(kusto_config)
```

Comment:
I'm more of a fan of the previous mode because it doesn't lose current `self` bindings

Reply:
(and if the caller wants empty, they can start with an empty)